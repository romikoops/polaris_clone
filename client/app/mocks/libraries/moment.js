import moment from 'moment'

require('moment-timezone')

moment.tz.setDefault('Europe/Amsterdam')
