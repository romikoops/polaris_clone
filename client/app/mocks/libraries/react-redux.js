import * as ReactRedux from 'react-redux'

ReactRedux.connect = jest.fn().mockImplementation((mapStateToProps, mapDispatchToProps) => Component => Component)
