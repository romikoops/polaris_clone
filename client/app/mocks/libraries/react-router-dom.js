import * as ReactRouterDom from 'react-router-dom'

ReactRouterDom.withRouter = jest.fn().mockImplementation(x => x)
