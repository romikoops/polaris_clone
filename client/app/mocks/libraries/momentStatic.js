import moment from 'moment'

require('moment-timezone')

const constantDate = new Date('June 13, 2017 04:41:20 GMT+1:00')

// eslint-disable-next-line
Date = class extends Date {
  constructor () {
    super()

    return constantDate
  }
}

moment.tz.setDefault('Europe/Amsterdam')
