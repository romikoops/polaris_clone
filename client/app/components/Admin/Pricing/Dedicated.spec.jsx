import '../../../mocks/libraries/moment'
import * as React from 'react'
import { shallow } from 'enzyme'
import { theme, identity } from '../../../mock'

import AdminPricingDedicated from './Dedicated'

const propsBase = {
  charges: [{
    transport_category: 'TRANSPORT',
    pricing: {
      data: {
        foo: { range: true },
        bar: {}
      }
    }
  }],
  clients: [],
  adminDispatch: {},
  theme,
  backBtn: identity,
  closePricingView: identity,
  initialEdit: false
}

test('shallow render', () => {
  expect(shallow(<AdminPricingDedicated {...propsBase} />)).toMatchSnapshot()
})

test('charges is falsy', () => {
  const props = {
    ...propsBase,
    charges: null
  }
  expect(shallow(<AdminPricingDedicated {...props} />)).toMatchSnapshot()
})

test('state.setUsers is true', () => {
  const charge = {
    transport_category: {
      cargo_class: {}
    },
    pricing: {
      data: {}
    }
  }
  const wrapper = shallow(<AdminPricingDedicated {...propsBase} />)

  wrapper.setState({ setUsers: true, charges: [charge] })

  expect(wrapper).toMatchSnapshot()
})

test('state.setUsers is false', () => {
  const charge = {
    transport_category: {
      cargo_class: {}
    },
    pricing: {
      data: {}
    }
  }
  const wrapper = shallow(<AdminPricingDedicated {...propsBase} />)

  wrapper.setState({ setUsers: false, charges: [charge] })

  expect(wrapper).toMatchSnapshot()
})
