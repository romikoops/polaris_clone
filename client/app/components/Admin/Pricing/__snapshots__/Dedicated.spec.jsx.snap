// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`charges is falsy 1`] = `""`;

exports[`shallow render 1`] = `""`;

exports[`state.setUsers is false 1`] = `
<div
  className="flex-100 layout-row layout-align-start-start layout-wrap container"
>
  <div
    className="flex-100 layout-row layout-align-space-between-center layout-wrap"
  >
    <div
      className="flex-5 layout-row pointy"
      onClick={[Function]}
    >
      <span
        className="hover_text"
      >
        &lt; Back
      </span>
    </div>
    <div
      className="flex-100 layout-row"
    >
      <h2>
        Define dedicated charges
      </h2>
    </div>
    <div
      className="flex-100 layout-row layout-align-start-start layout-wrap edit_wrapper"
    >
      <div
        className="flex-100 layout-row cargo_class_row"
      >
        <div
          className="flex-25 layout-row layout-align-start-center cargo_class_button_inactive cargo_class_button"
          onClick={null}
          style={
            Object {
              "background": "-webkit-linear-gradient(-90deg, #333,#fafafa)",
            }
          }
        >
          <div
            className="flex-none layout-row layout-align-center-center cargo_class_button_selected cargo_class_button_inner"
          >
            <p
              className="flex-none"
            >
              LCL
            </p>
          </div>
          <div
            className="flex-none cargo_class_divider"
          />
        </div>
        <div
          className="flex-25 layout-row layout-align-start-center cargo_class_button_inactive cargo_class_button"
          onClick={null}
          style={
            Object {
              "background": "#E0E0E0",
            }
          }
        >
          <div
            className="flex-none layout-row layout-align-center-center  cargo_class_button_inner"
          >
            <p
              className="flex-none"
            >
              FCL 10
            </p>
          </div>
          <div
            className="flex-none cargo_class_divider"
          />
        </div>
        <div
          className="flex-25 layout-row layout-align-start-center cargo_class_button_inactive cargo_class_button"
          onClick={null}
          style={
            Object {
              "background": "#E0E0E0",
            }
          }
        >
          <div
            className="flex-none layout-row layout-align-center-center  cargo_class_button_inner"
          >
            <p
              className="flex-none"
            >
              FCL 20ft
            </p>
          </div>
          <div
            className="flex-none cargo_class_divider"
          />
        </div>
        <div
          className="flex-25 layout-row layout-align-start-center cargo_class_button_inactive cargo_class_button"
          onClick={null}
          style={
            Object {
              "background": "#E0E0E0",
            }
          }
        >
          <div
            className="flex-none layout-row layout-align-center-center  cargo_class_button_inner"
          >
            <p
              className="flex-none"
            >
              FCL 20 Open Top
            </p>
          </div>
          <div
            className="flex-none cargo_class_divider"
          />
        </div>
        <div
          className="flex-25 layout-row layout-align-start-center cargo_class_button_inactive cargo_class_button"
          onClick={null}
          style={
            Object {
              "background": "#E0E0E0",
            }
          }
        >
          <div
            className="flex-none layout-row layout-align-center-center  cargo_class_button_inner"
          >
            <p
              className="flex-none"
            >
              FCL 20 Reefer
            </p>
          </div>
          <div
            className="flex-none cargo_class_divider"
          />
        </div>
        <div
          className="flex-25 layout-row layout-align-start-center cargo_class_button_inactive cargo_class_button"
          onClick={null}
          style={
            Object {
              "background": "#E0E0E0",
            }
          }
        >
          <div
            className="flex-none layout-row layout-align-center-center  cargo_class_button_inner"
          >
            <p
              className="flex-none"
            >
              FCL 20 Flat Rack (Sides)
            </p>
          </div>
          <div
            className="flex-none cargo_class_divider"
          />
        </div>
        <div
          className="flex-25 layout-row layout-align-start-center cargo_class_button_inactive cargo_class_button"
          onClick={null}
          style={
            Object {
              "background": "#E0E0E0",
            }
          }
        >
          <div
            className="flex-none layout-row layout-align-center-center  cargo_class_button_inner"
          >
            <p
              className="flex-none"
            >
              FCL 20 Flat Rack
            </p>
          </div>
          <div
            className="flex-none cargo_class_divider"
          />
        </div>
        <div
          className="flex-25 layout-row layout-align-start-center cargo_class_button_inactive cargo_class_button"
          onClick={null}
          style={
            Object {
              "background": "#E0E0E0",
            }
          }
        >
          <div
            className="flex-none layout-row layout-align-center-center  cargo_class_button_inner"
          >
            <p
              className="flex-none"
            >
              FCL 40ft
            </p>
          </div>
          <div
            className="flex-none cargo_class_divider"
          />
        </div>
        <div
          className="flex-25 layout-row layout-align-start-center cargo_class_button_inactive cargo_class_button"
          onClick={null}
          style={
            Object {
              "background": "#E0E0E0",
            }
          }
        >
          <div
            className="flex-none layout-row layout-align-center-center  cargo_class_button_inner"
          >
            <p
              className="flex-none"
            >
              FCL 40 Open Top
            </p>
          </div>
          <div
            className="flex-none cargo_class_divider"
          />
        </div>
        <div
          className="flex-25 layout-row layout-align-start-center cargo_class_button_inactive cargo_class_button"
          onClick={null}
          style={
            Object {
              "background": "#E0E0E0",
            }
          }
        >
          <div
            className="flex-none layout-row layout-align-center-center  cargo_class_button_inner"
          >
            <p
              className="flex-none"
            >
              FCL 40 Reefer
            </p>
          </div>
          <div
            className="flex-none cargo_class_divider"
          />
        </div>
        <div
          className="flex-25 layout-row layout-align-start-center cargo_class_button_inactive cargo_class_button"
          onClick={null}
          style={
            Object {
              "background": "#E0E0E0",
            }
          }
        >
          <div
            className="flex-none layout-row layout-align-center-center  cargo_class_button_inner"
          >
            <p
              className="flex-none"
            >
              FCL 40 Flat Rack (Sides)
            </p>
          </div>
          <div
            className="flex-none cargo_class_divider"
          />
        </div>
        <div
          className="flex-25 layout-row layout-align-start-center cargo_class_button_inactive cargo_class_button"
          onClick={null}
          style={
            Object {
              "background": "#E0E0E0",
            }
          }
        >
          <div
            className="flex-none layout-row layout-align-center-center  cargo_class_button_inner"
          >
            <p
              className="flex-none"
            >
              FCL 40 Flat Rack
            </p>
          </div>
          <div
            className="flex-none cargo_class_divider"
          />
        </div>
        <div
          className="flex-25 layout-row layout-align-start-center cargo_class_button_inactive cargo_class_button"
          onClick={null}
          style={
            Object {
              "background": "#E0E0E0",
            }
          }
        >
          <div
            className="flex-none layout-row layout-align-center-center  cargo_class_button_inner"
          >
            <p
              className="flex-none"
            >
              FCL 40ft HQ
            </p>
          </div>
          <div
            className="flex-none cargo_class_divider"
          />
        </div>
        <div
          className="flex-25 layout-row layout-align-start-center cargo_class_button_inactive cargo_class_button"
          onClick={null}
          style={
            Object {
              "background": "#E0E0E0",
            }
          }
        >
          <div
            className="flex-none layout-row layout-align-center-center  cargo_class_button_inner"
          >
            <p
              className="flex-none"
            >
              FCL 45
            </p>
          </div>
          <div
            className="flex-none cargo_class_divider"
          />
        </div>
        <div
          className="flex-25 layout-row layout-align-start-center cargo_class_button_inactive cargo_class_button"
          onClick={null}
          style={
            Object {
              "background": "#E0E0E0",
            }
          }
        >
          <div
            className="flex-none layout-row layout-align-center-center  cargo_class_button_inner"
          >
            <p
              className="flex-none"
            >
              FCL 45 HQ
            </p>
          </div>
          <div
            className="flex-none cargo_class_divider"
          />
        </div>
        <div
          className="flex-25 layout-row layout-align-start-center cargo_class_button_inactive cargo_class_button"
          onClick={null}
          style={
            Object {
              "background": "#E0E0E0",
            }
          }
        >
          <div
            className="flex-none layout-row layout-align-center-center  cargo_class_button_inner"
          >
            <p
              className="flex-none"
            >
              FCL 45 Reefer
            </p>
          </div>
        </div>
      </div>
      <div
        className="flex-100 layout-row layout-align-start-start layout-wrap fee_row_container"
      />
    </div>
    <div
      className="flex-100 layout-row layout-align-end-center layout-wrap fee_row_container"
    >
      <div
        className="flex-33 layout-row"
      >
        <RoundButton
          active={true}
          back={false}
          classNames=""
          disabled={false}
          flexContainer="100"
          handleDisabled={null}
          handleNext={[Function]}
          icon=""
          iconClass=""
          inverse={true}
          size="small"
          text="Next >"
          theme={
            Object {
              "colors": Object {
                "primary": "#333",
                "secondary": "#fafafa",
              },
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`state.setUsers is true 1`] = `
<div
  className="flex-100 layout-row layout-align-start-start layout-wrap container"
>
  <div
    className="flex-100 layout-row layout-align-space-between-center layout-wrap"
  >
    <div
      className="flex-5 layout-row pointy"
      onClick={[Function]}
    >
      <span
        className="hover_text"
      >
        &lt; Back
      </span>
    </div>
    <div
      className="flex-100 layout-row layout-align-space-between-center"
    >
      <h2>
        Choose users
      </h2>
      <div
        className="input_box_full flex-40 layout-row layout-align-end-center"
      >
        <input
          name="search"
          onChange={[Function]}
          placeholder="Search clients"
          type="text"
        />
      </div>
    </div>
    <div
      className="flex-100 layout-row layout-align-start-start layout-wrap layout-padding users_wrapper"
    />
    <div
      className="flex-100 layout-row layout-align-center-center layout-wrap"
    >
      <RoundButton
        active={true}
        back={false}
        classNames=""
        disabled={false}
        flexContainer="100"
        handleDisabled={null}
        handleNext={[Function]}
        icon=""
        iconClass=""
        inverse={false}
        size="small"
        text="Save Pricings"
        theme={
          Object {
            "colors": Object {
              "primary": "#333",
              "secondary": "#fafafa",
            },
          }
        }
      />
    </div>
  </div>
</div>
`;
